src/Compilers/CSharp/Test/Semantic/Semantics/NullConditionalAssignmentTests.cs

```diff
@@ -0,0 +1,1762 @@
+                  IL_0039:  ret
+                }
+                """);
+
+            verifier.VerifyIL("S.M2", """
+                {
+                  // Code size       37 (0x25)
+                  .maxstack  3
+                  .locals init (int V_0,
+                                S V_1)
+                  IL_0000:  ldarga.s   V_0
+                  IL_0002:  call       "bool S?.HasValue.get"
+                  IL_0007:  brtrue.s   IL_000c
+                  IL_0009:  ldc.i4.4
+                  IL_000a:  br.s       IL_001f
+                  IL_000c:  ldarga.s   V_0
+                  IL_000e:  call       "S S?.GetValueOrDefault()"
+                  IL_0013:  stloc.1
+                  IL_0014:  ldloca.s   V_1
+                  IL_0016:  ldc.i4.2
+                  IL_0017:  dup
+                  IL_0018:  stloc.0
+                  IL_0019:  call       "readonly void S.RP.set"
+                  IL_001e:  ldloc.0
+                  IL_001f:  call       "void System.Console.Write(int)"
+                  IL_0024:  ret
+                }
+                """);
+
+            verifier.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DeconstructionLeft()
+        {
+            var source = """
+                using System;
+
+                class C
+                {
+                    int F;
+
+                    static void M1(C c1, C c2)
+                    {
+                        (c1?.F, c2?.F) = (1, 2); // 1
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (5,9): warning CS0649: Field 'C.F' is never assigned to, and will always have its default value 0
+                //     int F;
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "F").WithArguments("C.F", "0").WithLocation(5, 9),
+                // (9,10): error CS0131: The left-hand side of an assignment must be a variable, property or indexer
+                //         (c1?.F, c2?.F) = (1, 2); // 1
+                Diagnostic(ErrorCode.ERR_AssgLvalueExpected, "c1?.F").WithLocation(9, 10),
+                // (9,17): error CS0131: The left-hand side of an assignment must be a variable, property or indexer
+                //         (c1?.F, c2?.F) = (1, 2); // 1
+                Diagnostic(ErrorCode.ERR_AssgLvalueExpected, "c2?.F").WithLocation(9, 17));
+        }
+
+        [Fact]
+        public void RefAssignment_01()
+        {
+            var source = """
+                using System;
+
+                ref struct RS
+                {
+                    ref int RF;
+
+                    static void M1()
+                    {
+                        int i = 0;
+                        var rs = new RS { RF = ref i };
+                        rs?.RF = ref i; // 1
+
+                        RS? nrs = rs; // 2
+                        nrs?.RF = ref i; // 3
```

Comment:
That's an interesting thought, as the assignment itself is fine. Left side doesn't need to be a variable in this case. It just doesn't work because a ref struct can't be nullable.